File: src/main/java/com/example/backendthuvien/DTO/OrderDTO.java
package com.example.backendthuvien.DTO;

import com.example.backendthuvien.entity.CartItems;
import com.example.backendthuvien.entity.Coupon;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.Column;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.*;

import java.time.LocalDate;
import java.util.Date;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@ToString
public class OrderDTO {

    @JsonProperty("fullname")
    private String fullName;

    private String email;

    @JsonProperty("phone_number")
    @NotBlank(message = "khong de trong")
    @Size(min = 5,message = "SDT khong hop le")
    private String phoneNumber;

    private String address;

    private String note;

    @JsonProperty("total_money")
    @Min(value = 0,message = "Gia tien lon hon or bang 0")
    private float totalMoney;

    @JsonProperty("shipping_method")
    private String shippingMethod;

    @JsonProperty("shipping_address")
    private String shippingAddress;


    @JsonProperty("payment_method")
    private String paymentMethod;

    @JsonProperty("shipping_date")
    private LocalDate shippingDate;

    @Min(value = 1,message = "Id > 0 nha")
    @JsonProperty("user_id")
    private Long userId;

    @JsonProperty("active")
    private Boolean active;//thuộc về admin

@JsonProperty("cart_items")
    private List<CartItemsDTO> cartItems;

    @JsonProperty("coupon_code")
    private String couponCode;


    private String orderType; // 'purchase' hoặc 'borrow'


    private LocalDate borrowDate; // Ngày mượn sách (nếu là borrow)


    private LocalDate returnDate; // Ngày trả sách (nếu là borrow)


    private String borrowStatus; // Trạng thái đơn mượn (pending, completed, overdue)








}

File: src/app/responses/user/orderDTO.ts
import { OrderDetailDTO } from "./orderDetailDTO";

export class OrderDTO {
  id: number;
  user_id: number;           
  fullname: string;          
  email: string;             
  phone_number: string;      
  address: string;           
  note: string;              
  total_money: number;       
  shipping_method: string;   
  payment_method: string;    
  coupon_code: string; 
  order_date: Date;      
  cart_items: {              
    product_id: number;      
    quantity: number;        
  }[];
  active: string;  // Thêm thuộc tính active
  orderDetail: OrderDetailDTO[] = [];
  constructor(data: any) {
    this.id=data.id;
    this.user_id = data.user_id;
    this.fullname = data.fullname;
    this.email = data.email;
    this.phone_number = data.phone_number;
    this.address = data.address;
    this.note = data.note;
    this.total_money = data.total_money;
    this.shipping_method = data.shipping_method;
    this.payment_method = data.payment_method;
    this.coupon_code = data.coupon_code || '';
    this.order_date=data.order_date;
    this.cart_items = data.cart_items || [];
    this.active = data.active || 'Đang hoạt động'; // Giá trị mặc định
    this.orderDetail = data.orderDetails || [];
  }
}
